src/crypto/random.c

```diff
@@ -146,3 +146,18 @@ void generate_random_bytes_not_thread_safe(size_t n, void *result) {
     }
   }
 }
+
+void add_extra_entropy(const void *ptr, size_t bytes)
+{
+  size_t i;
+
+  while (bytes > 0)
+  {
+    hash_permutation(&state);
+    const size_t round_bytes = bytes > HASH_SIZE ? HASH_SIZE : bytes;
```

Comment:
`HASH_SIZE` -> `HASH_DATA_AREA` which is the exact number of bytes (for one of the) specified implementations in the Keccak paper, and the amount used when retrieving ("squeezing") bytes. Although, `HASH_SIZE` should work fine, but does less bytes per iteration.